txstratum/cli.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Hathor Labs and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import asyncio
+import logging
+import logging.config
+import os
+from argparse import ArgumentParser, Namespace
+
+import structlog  # type: ignore
+from aiohttp import web
+from structlog import get_logger
+from txstratum.api import App
+from txstratum.manager import TxMiningManager
+
+logger = get_logger()
+
+
+def create_parser() -> ArgumentParser:
+    """Create a parser for the cmdline arguments."""
+    import configargparse  # type: ignore
+    parser: ArgumentParser = configargparse.ArgumentParser(auto_env_var_prefix='hathor_')
+    parser.add_argument('--stratum-port', help='Port of Stratum server', type=int, default=8000)
+    parser.add_argument('--api-port', help='Port of TxMining API server', type=int, default=8080)
+    parser.add_argument('--log-config', help='Config file for logging', default='log.conf')
+    parser.add_argument('--max-tx-weight', help='Maximum allowed tx weight to be mined', type=int, default=None)
```

Comment:
I added both `--max-tx-weight` and `--tx-timeout`. @jansegre @obiyankenobi @pedroferreira1 Would you please review it? I put in a separate commit to make it easier to see the changes.

Reply:
I don't think it's super important for `max-tx-weight` to be configurable, but it's fine. But I think it should have a reasonable default. It certainly doesn't make sense to allow impossibly large tx.weights. I'd be OK if we just estimated an upper bound for tx.weight and made that hardcoded.

`--tx-timeout` looks great though.

Reply:
I just noticed that `--max-tx-weight` defaults to `MAX_TX_WEIGHT` when it's `None`. I think it's fine then.

Reply:
Ok!